Jetsnack/app/src/main/java/com/example/jetsnack/ui/JetsnackApp.kt

```diff
@@ -41,15 +40,6 @@ fun JetsnackApp() {
     JetsnackTheme {
         val appState = rememberJetsnackAppState()
         JetsnackScaffold(
-            bottomBar = {
-                if (appState.shouldShowBottomBar) {
-                    JetsnackBottomBar(
-                        tabs = appState.bottomBarTabs,
-                        currentRoute = appState.currentRoute!!,
-                        navigateToRoute = appState::navigateToBottomBarRoute
-                    )
-                }
-            },
             snackbarHost = {
                 SnackbarHost(
```

Comment:
The Snackbars are now obscuring the content of BottomBar, I think it would be good to include the SnackbarHost per Scaffold too, where required.

Reply:
Good catch! It's because scaffoldState is managed at UI tree root level (JetsnackApp) and we removed the bottom bar there. In my last commit I'm using `rememberJetsnackAppState` in Cart composable to access scaffoldSate, I think is a bit overkill as I don't need access to the navController (anyway it's solving the problem). I also don't like passing the scaffoldState down the UI tree. Maybe I could refactor `JetsnackAppState` removing scaffoldState from it and create a custom `JetsnackScaffoldState`. Something like this, what do you think?
```
/**
 * Remember and created an instance of [JetsnackScaffoldState]
 */
@Composable
fun rememberJetsnackScaffoldState(
    scaffoldState: ScaffoldState = rememberScaffoldState(),
    snackbarManager: SnackbarManager = SnackbarManager,
    resources: Resources = resources(),
    coroutineScope: CoroutineScope = rememberCoroutineScope()
    ): JetsnackScaffoldState = remember(scaffoldState, snackbarManager, resources, coroutineScope) {
        JetsnackScaffoldState(scaffoldState, snackbarManager, resources, coroutineScope)
}

/**
 * Responsible for holding [ScaffoldState], handles the logic of showing snackbar messages
 */
@Stable
class JetsnackScaffoldState(
    val scaffoldState: ScaffoldState,
    private val snackbarManager: SnackbarManager,
    private val resources: Resources,
    coroutineScope: CoroutineScope
) {
    // Process snackbars coming from SnackbarManager
    init {
        coroutineScope.launch {
            snackbarManager.messages.collect { currentMessages ->
                if (currentMessages.isNotEmpty()) {
                    val message = currentMessages[0]
                    val text = resources.getText(message.messageId)
                    // Notify the SnackbarManager so it can remove the current message from the list
                    snackbarManager.setMessageShown(message.id)
                    // Display the snackbar on the screen. `showSnackbar` is a function
                    // that suspends until the snackbar disappears from the screen
                    scaffoldState.snackbarHostState.showSnackbar(text.toString())
                }
            }
        }
    }
}
```

Reply:
Yeah the proposed change will be better I agree.